### ISSUE TITLE
Graph visualization doesn't respect query range defined in flux query

### ISSUE THREAD

**Issue by genehynson** [influxdata/ui#5890]
Query:

```
from(bucket: "iot")
  |> range(start: 2022-09-21T15:26:00Z, stop: 2022-09-21T15:30:00Z)
  |> filter(fn: (r) => r["_measurement"] == "obd")
  |> filter(fn: (r) => r["_field"] == "mph")
  |> toFloat()
```

Graph doesn't respect the range:

<img width="1538" alt="image" src="https://user-images.githubusercontent.com/6411855/192343080-922c5202-47ee-469c-8ab7-fc472cb581bb.png">

Change query to use custom time range dropdown:

```
from(bucket: "iot")
  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
  |> filter(fn: (r) => r["_measurement"] == "obd")
  |> filter(fn: (r) => r["_field"] == "mph")
  |> toFloat()
```

And now the visualization is correct:

<img width="1546" alt="image" src="https://user-images.githubusercontent.com/6411855/192343291-91d3873f-a9e2-462e-9642-97c3a21cbe47.png">

## About your environment

**Environment info:**
- https://us-west-2-1.aws.cloud2.influxdata.com/

**Comment by kristinarobinson** [comment]
@TCL735, @hralexander -- I know that Denise and Timmy both looked at the query range when the dropdown is used or when hardcoded within the flux.  Can either or both of you see if you can repro, and hand off to the appropriate owner.

**Comment by cannonpalms** [comment]
## Hard-coded time range in flux query

On https://go.influxcorp.co/newtools
```flux
from(bucket: "query_log")
  |> range(start: -24h)
  |> filter(fn: (r) => r["_measurement"] == "query_log")
  |> filter(fn: (r) => r["_field"] == "executeDuration")
  |> filter(fn: (r) => r["env"] == "prod01-eu-central-1")
  |> filter(fn: (r) => r["orgID"] == "2bc31ebe26081a78")
  |> filter(fn: (r) => r["source"] == "influxdb-client-go")
  |> filter(fn: (r) => r["role"] == "queryd-pull-external")
  |> group()
  |> map(fn: (r) => ({ r with _value: float(v: r._value) / float(v: int(v: 1s))}))
  |> aggregateWindow(every: 5m, fn: (tables=<-, column) => tables
      |> quantile(q: 0.99), createEmpty: false)
      |> exponentialMovingAverage(n: 200)
```

### With adaptive zoom `ON`: 
![image (4)](https://user-images.githubusercontent.com/2457559/193349703-8ce87669-b5b6-48c9-8034-ece76cd82d4e.png)

### With adaptive zoom `OFF`:
![image (5)](https://user-images.githubusercontent.com/2457559/193349743-ce7b8cb8-c6c6-4562-95c1-ee0f87d56ec2.png)

## Using time range from drop down

```flux
from(bucket: "query_log")
  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
  |> filter(fn: (r) => r["_measurement"] == "query_log")
  |> filter(fn: (r) => r["_field"] == "executeDuration")
  |> filter(fn: (r) => r["env"] == "prod01-eu-central-1")
  |> filter(fn: (r) => r["orgID"] == "2bc31ebe26081a78")
  |> filter(fn: (r) => r["source"] == "influxdb-client-go")
  |> filter(fn: (r) => r["role"] == "queryd-pull-external")
  |> group()
  |> map(fn: (r) => ({ r with _value: float(v: r._value) / float(v: int(v: 1s))}))
  |> aggregateWindow(every: 5m, fn: (tables=<-, column) => tables
      |> quantile(q: 0.99), createEmpty: false)
      |> exponentialMovingAverage(n: 200)
```

### With adaptive zoom `ON`: 
![image](https://user-images.githubusercontent.com/2457559/193350007-604511b8-a604-4881-aec0-f4c3acd24949.png)

### With adaptive zoom `OFF`: 
![image](https://user-images.githubusercontent.com/2457559/193350007-604511b8-a604-4881-aec0-f4c3acd24949.png)

**Comment by TCL735** [comment]
The adaptive zoom issues should be fixed with https://github.com/influxdata/ui/pull/6048

**Comment by TCL735** [comment]
There is possibly a domain issue unrelated to adaptive zoom, and that is the user not wanting the dropdown to interfere with a written Flux query that contains a specific time range. It seems like there has always been this disconnect of the user writing a Flux query with a specific time range and then getting frustrated at seeing on a slice of that data, or vice versa -- a specific time range (small) that also includes a large empty domain in the visualization.

To resolve this, the time range dropdown should have another option that is basically equivalent to "just show me whatever my Flux gave back". Currently, the custom time range fills this role. However, this can be an onerous task because the user is required to specify their time range twice (once in the Flux query, and again in the time range dropdown).

Per @kristinarobinson we'll check internally with stakeholders before adding the above option ("just show me whatever my Flux gave me") to the time range dropdown.

**Comment by cannonpalms** [comment]
I'd argue that the drop-down should be completely ignored if the user provides an explicit time range in the query. 

On another note, relating to both zoom and the drop-down, why does zooming not change the time range to custom with the appropriate bounds automatically? This is how grafana behaves and I quite like it there. However, grafana has the advantage of embedding the query metadata (like the time range) and query text in the query parameters of the current URL. This makes undoing the change that zooming makes to the currently configured time range as simple as navigating back with browser history. 

This is a bit off topic for this issue, though. Is there somewhere else we can continue this discussion?

**Comment by kristinarobinson** [comment]
OK, after meeting together, we've decided that this issue will be addressed by the QX team, when they overhaul the time duration selection process.  Too many edge cases exist for a patch to be put in place at this team.

**Comment by TCL735** [comment]
Also, I wanted to clear up what the actual problem is. The time range dropdown isn't overriding the time domain necessarily. What it is really doing is adding extra space to the time domain whether there is actually any data there or not.

For example, if a user has in their Flux `start: -7d, stop: -5d` but the dropdown is selected as `Past 7d` then the visualization will include the entire domain of the past 7 days, instead of the slice of time between 5 days ago and 7 days ago.

And this can also happen on the other end of the domain.

To summarize, the time range dropdown is affecting the minimum and maximum time domain when the user might not want it to do so.
